### PR TITLE
PWX-33964: Resize pool with pending pool operation

### PR DESCRIPTION
**Pull request by alicelyu-px**
Successful runs: 
https://jenkins.pwx.dev.purestorage.com/job/Users/job/alyu/job/pxdv-7-dev/88/
https://jenkins.pwx.dev.purestorage.com/job/Users/job/alyu/job/pxdv-7-dev/89/

This test is on top of https://github.com/portworx/torpedo/pull/1767

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -214,6 +215,30 @@ var _ = Describe("{PoolExpandRejectConcurrent}", func() {
...
+			log.Fatalf("Error checking if pool expansion is in progress: %v", expandErr)
+		}
+		if !isExpandInProgress {
+			log.Warnf("Pool expansion already finished. Skipping test. Please retry with an app that writes more data to testing volumes. ")
+			return
+		}
+		expandResponse := Inst().V.ExpandPoolUsingPxctlCmd(*storageNode, poolToResize.Uuid, expandType, targetSize+100, true)
+		dash.VerifyFatal(expandResponse != nil, true, "Pool expansion should fail when expansion is in progress")
+		dash.VerifyFatal(strings.Contains(expandResponse.Error(), "is already in progress"), true,
+			"Pool expansion failure reason should be communicated to the user	")
```
Comment (pp511) [inline tests/basic/pool_function_test.go:241]: nit.. Why too many spaces towards the end of the statement?

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -214,6 +215,30 @@ var _ = Describe("{PoolExpandRejectConcurrent}", func() {
...
+		// TODO: this is a hack to wait for expansion to start. The existing WaitForExpansionToStart() risks returning
+		// when the expansion has already completed.
+		time.Sleep(1)
+		// verify pool expansion is in progress
+		isExpandInProgress, expandErr := poolResizeIsInProgress(poolToResize)
+		if expandErr != nil {
+			log.Fatalf("Error checking if pool expansion is in progress: %v", expandErr)
+		}
+		if !isExpandInProgress {
+			log.Warnf("Pool expansion already finished. Skipping test. Please retry with an app that writes more data to testing volumes. ")
```
Comment (pp511) [inline tests/basic/pool_function_test.go]: nit..Why a space at the end?

File: tests/basic/pool_function_test.go [hunk 3]
```diff
@@ -214,6 +215,30 @@ var _ = Describe("{PoolExpandRejectConcurrent}", func() {
...
+			return
+		}
+		expandResponse := Inst().V.ExpandPoolUsingPxctlCmd(*storageNode, poolToResize.Uuid, expandType, targetSize+100, true)
```
Comment (pp511) [inline tests/basic/pool_function_test.go:238]: I am not aware if there is a limitation the restricts us to use ExpandPool again here. Can you explain why did we have to use ExpandPoolUsingPxctlCmd here?

**Review by pp511 — APPROVED** [review]
lgtm. Just need you to answer one q and a nit.

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (alicelyu-px) [reply tests/basic/pool_function_test.go]: Resolved offline

**Pull request merged**